### PR TITLE
Include screen to create account

### PR DESCRIPTION
**Pull request by hvhallmann**
Include screen for user create account

## Changes

- Include async storage
- Include method to include record on async storage
- create new screen 
- validate user input

### New Features *(required)*

Async storage

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: src/CreateAccount.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  const [username, setUsername] = useState("");
+  const [error, setError] = useState(false);
+
+  const onSubmit = async () => {
+    const userData = await Motor.createAccount(username, vaultPassword);
+    if (!userData) {
+      setError(true);
+      return;
+    }
+    alert("Account created");
```
[the remark this answers is not in the document]
Comment (hvhallmann) [reply src/CreateAccount.tsx]: I changed this code a bit. Now it moves to the next screen. It seems the user could set device name.
You think it should get back to Beam app, calling onSucces and close?

File: src/CreateAccount.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+          value={username}
+          onChangeText={setUsername}
+          warning={invalidInput}
+          icon="user"
+        />
+
+        <IconText
+          label="Your Vault Password"
+          value={vaultPassword}
+          onChangeText={(newText) => setVaultPassword(newText)}
```
Comment (tgfassina) [inline src/CreateAccount.tsx:62]: I think it can be just a `setVaultPassword`, same as the previous input

**Comment by tgfassina** [comment]
Also, for some reason, when I go into "Create Account" my cursor goes directly to the password field (not great)

File: src/AccountCreated.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      <View style={styles.header}>
+        <SonrLogo />
+        <Text style={styles.close} onPress={() => {}}>
+          Close
+        </Text>
+      </View>
+
+      <View style={styles.content}>
+        <Text style={styles.subtitle2}>Account Created!</Text>
+        <WalletAddress
```
Comment (tgfassina) [inline src/AccountCreated.tsx:31]: Lets leave this part about devices hidden for now. I don't think we're gonna need it on first version

File: src/CreateAccount.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (tgfassina) [reply src/CreateAccount.tsx]: It's nice that we have the next screen! Then we have to see if it makes more sense to do the success callback from here or on the next screen. (I don't know)

[the remark this answers is not in the document]
Comment (hvhallmann) [reply src/CreateAccount.tsx]: ok, I put on next screen, not sure also, if that will make a difference for now

File: src/CreateAccount.tsx (hunk 2 above)
Comment (hvhallmann) [reply src/CreateAccount.tsx:62]: right

File: src/sandbox/index.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import AsyncStorage from "@react-native-async-storage/async-storage";
+
 const database = [
```
Comment (tgfassina) [inline src/sandbox/index.ts:3]: @hvhallmann We're no longer using this, right?

Comment (tgfassina) [reply src/sandbox/index.ts:3]: I think it can go

Comment (hvhallmann) [reply src/sandbox/index.ts:3]: I'm using to generate a first input.
With it we can insert matrix know passwords,
not sure if needed though

Comment (tgfassina) [reply src/sandbox/index.ts:3]: Don't need the extra complexity now that we can register

Comment (tgfassina) [reply src/sandbox/index.ts:3]: don't worry. I get it

**Pull request merged**